lib/logstash/outputs/elasticsearch/template_manager.rb

```diff
@@ -7,15 +7,15 @@ class TemplateManager
     def self.install_template(plugin)
       return unless plugin.manage_template
 
-      if plugin.maximum_seen_major_version < 8 && plugin.template_api == 'auto'
-        plugin.logger.warn("`template_api => auto` resolved to `legacy` since we are connected to " + "Elasticsearch #{plugin.maximum_seen_major_version}, " +
-                           "but will resolve to `composable` the first time it connects to Elasticsearch 8+. " +
-                           "We recommend either setting `template_api => legacy` to continue providing legacy-style templates, " +
-                           "or migrating your template to the composable style and setting `template_api => composable`. " +
-                           "The legacy template API is slated for removal in Elasticsearch 9.")
-      end
-
       if plugin.template
+        if plugin.maximum_seen_major_version < 8 && plugin.template_api == 'auto'
```

Comment:
review note: since `template_api` is a hint about `template`, only check-and-emit this warning if the user has provided a `template`